src/firmware_new/nanopb.bzl

```diff
@@ -0,0 +1,130 @@
+load("@rules_proto//proto:defs.bzl", "ProtoInfo")
+load("@rules_cc//cc:toolchain_utils.bzl", "find_cpp_toolchain")
+
+def _nanopb_proto_library_impl(ctx):
+    # This is the folder we will place all our generation artifacts in
+    generation_folder_name = ctx.attr.name + "_nanopb_gen/"
+
+    # Generate import flags for the protobuf compiler so it can find proto files we
+    # depend on, and a list of proto files to include
+    all_proto_files = depset()
+    for dep in ctx.attr.deps:
+        all_proto_files = depset(
+            transitive = [all_proto_files, dep[ProtoInfo].transitive_sources],
+        )
+
+    all_proto_hdr_files = []
+    all_proto_src_files = []
+
+    for proto_file in all_proto_files.to_list():
+        # Generate the `.pb` files using the protobuf compiler
+        pb_file_name = generation_folder_name + proto_file.path[:-len(".proto")] + ".pb"
+        pb_file = ctx.actions.declare_file(pb_file_name)
+
+        ctx.actions.run(
+            inputs = all_proto_files,
+            outputs = [pb_file],
+            arguments = ["-o", pb_file.path, proto_file.path],
+            executable = ctx.executable.protoc,
+            mnemonic = "ProtoCompile",
+            use_default_shell_env = True,
+        )
+
+        # Generate the equivalent C code using Nanopb
+        h_out_name = generation_folder_name + proto_file.path[:-len(".proto")] + ".pb.h"
```

Comment:
`udp_multicast` is currently expecting `_fw.pb.h` for nanopb generated headers, I think you are going to need to append something here, because both C and C++ generate `.pb.h` files

that also seems to be the reason for the CI failure

Reply:
Whoops. I'll look into this tomorrow, but off the top of my head I can't think of any reason any target would depend on both?

Reply:
Looks like the I just had to re-add `fw_cc_proto`. I think as long as no target depends on both the C and C++ protos at the same time, bazel will make sure there are never any header conflicts via sandboxing. :tada: